tests/e2e/stablehlo_ops/BUILD.bazel

```diff
@@ -89,7 +89,6 @@ iree_check_single_backend_test_suite(
         "--iree-llvmcpu-target-cpu=generic",
     ],
     driver = "local-task",
-    input_type = "stablehlo",
```

Comment:
There's some history here with https://github.com/iree-org/iree/pull/15378#discussion_r1379411505

See https://github.com/iree-org/iree/blob/5b154e5994f208c8bcb0a167e0d9c6583904b520/build_tools/cmake/iree_check_test.cmake#L72-L87

If we remove these explicit input types, building `iree-test-deps` with `-DIREE_INPUT_STABLEHLO=OFF` will fail. I _think_ that should currently work, but something else may have regressed.

Reply:
We could also keep the existing behavior by putting

```cmake
if(NOT IREE_HOST_BIN_DIR AND NOT IREE_INPUT_STABLEHLO)
  return()
endif()
```

at the top of [`tests/e2e/stablehlo_ops/CMakeLists.txt`](https://github.com/iree-org/iree/blob/main/tests/e2e/stablehlo_ops/CMakeLists.txt) then continue with the changes in this PR.

Same for e2e/tosa_ops, but that could be tricky for e2e/regression - maybe move those test cases to stablehlo_regression and tosa_regression...?

Reply:
I did not know the trick, but it looks good to me!

> maybe move those test cases to stablehlo_regression and tosa_regression...?

Yes, this is similar to what I'm thinking. I'm thinking to put additional folders within `regression/`, like `regression/stablehlo`, `regression/tosa`, `regression/linalg`, etc.

`linalg` is a bad choice to me because `linalg_ext` can be in the same folder. I'm looking for a terminology for `iree-input-type=none`, but I dont have it now.